app/models/concerns/course/levels_concern.rb

```diff
@@ -41,4 +41,34 @@ def compute_level_number(experience_points)
     level = compute_level(experience_points)
     level ? level.level_number : 0
   end
+
+  # Returns the current level's experience point (exp) threshold for
+  # for a given number of experience_points.
+  #
+  # 0 returned if levels are not present, or if exp is below threshold of first level
+  #
+  # @param [Fixnum] experience_points Number of experience points
+  # @return [Fixnum] The experience_points threshold of the current level
+  def current_level_threshold(experience_points)
+    level = compute_level(experience_points)
+    level ? level.experience_points_threshold : 0
+  end
+
+  # Returns the next highest level's experience point (exp) threshold for
+  # for a given number of experience_points.
+  #
+  # 0 returned if levels are not present
+  # x returned if exp exceeds threshold of highest level, where x is threshold of
+  # highest level
+  #
+  # @param [Fixnum] experience_points Number of experience points
+  # @return [Fixnum] The experience_points threshold of the next higher level
+  def next_level_threshold(experience_points)
+    return 0 unless numbered_levels
```

Comment:
@weiqingtoh Thanks ! 

One small issue, can `numbered_levels` be cached ?  you are computing the `numbered_levels` for 4 times in this method.